Mathlib/CategoryTheory/Comma/Basic.lean

```diff
@@ -196,6 +199,71 @@ def isoMk {X Y : Comma L₁ R₁} (l : X.left ≅ Y.left) (r : X.right ≅ Y.rig
         simp }
 #align category_theory.comma.iso_mk CategoryTheory.Comma.isoMk
 
+section
+
+variable {L R}
+variable {L' : A' ⥤ T'} {R' : B' ⥤ T'}
+  {F₁ : A ⥤ A'} {F₂ : B ⥤ B'} {F : T ⥤ T'}
+  (α : F₁ ⋙ L' ⟶ L ⋙ F) (β : R ⋙ F ⟶ F₂ ⋙ R')
+
+/-- The functor `Comma L R ⥤ Comma L' R'` induced by three functors `F₁`, `F₂`, `F`
+and two natural transformations `F₁ ⋙ L' ⟶ L ⋙ F` and `R ⋙ F ⟶ F₂ ⋙ R'`. -/
+@[simps]
+def map : Comma L R ⥤ Comma L' R' where
+  obj X :=
+    { left := F₁.obj X.left
+      right := F₂.obj X.right
+      hom := α.app X.left ≫ F.map X.hom ≫ β.app X.right }
+  map {X Y} φ :=
+    { left := F₁.map φ.left
+      right := F₂.map φ.right
+      w := by
+        dsimp
+        rw [assoc, assoc]
+        erw [α.naturality_assoc, ← β.naturality]
+        dsimp
+        rw [← F.map_comp_assoc, ← F.map_comp_assoc, φ.w] }
+
+instance faithful_map [F₁.Faithful] [F₂.Faithful] : (map α β).Faithful where
+  map_injective {X Y} f g h := by
+    ext
+    · exact F₁.map_injective (congr_arg CommaMorphism.left h)
+    · exact F₂.map_injective (congr_arg CommaMorphism.right h)
+
+instance fullMap [F.Faithful] [F₁.Full] [F₂.Full] [IsIso α] [IsIso β] : (map α β).Full where
+  preimage {X Y} φ :=
+    { left := F₁.preimage φ.left
+      right := F₂.preimage φ.right
+      w := F.map_injective (by
+        rw [← cancel_mono (β.app _), ← cancel_epi (α.app _), F.map_comp, F.map_comp,
+          assoc, assoc]
+        erw [← α.naturality_assoc, β.naturality]
+        dsimp
+        rw [F₁.image_preimage, F₂.image_preimage]
+        simpa using φ.w) }
+
+instance essSurj_map [F₁.EssSurj] [F₂.EssSurj] [F.Full] [IsIso α] [IsIso β] :
+    (map α β).EssSurj where
+  mem_essImage X :=
+    ⟨{  left := F₁.objPreimage X.left
+        right := F₂.objPreimage X.right
+        hom := F.preimage ((inv α).app _ ≫ L'.map (F₁.objObjPreimageIso X.left).hom ≫
+          X.hom ≫ R'.map (F₂.objObjPreimageIso X.right).inv ≫ (inv β).app _) },
+            ⟨isoMk (F₁.objObjPreimageIso X.left) (F₂.objObjPreimageIso X.right) (by
+              dsimp
+              simp only [NatIso.isIso_inv_app, Functor.comp_obj, Functor.image_preimage, assoc,
+                IsIso.inv_hom_id, comp_id, IsIso.hom_inv_id_assoc]
+              rw [← R'.map_comp, Iso.inv_hom_id, R'.map_id, comp_id])⟩⟩
+
+noncomputable instance isEquivalenceMap
+    [F₁.IsEquivalence] [F₂.IsEquivalence] [F.Faithful] [F.Full] [IsIso α] [IsIso β] :
+    (map α β).IsEquivalence := by
+  have := Equivalence.essSurj_of_equivalence F₁
+  have := Equivalence.essSurj_of_equivalence F₂
+  apply Functor.IsEquivalence.ofFullyFaithfullyEssSurj
+
+end
+
 /-- A natural transformation `L₁ ⟶ L₂` induces a functor `Comma L₂ R ⥤ Comma L₁ R`. -/
 @[simps]
 def mapLeft (l : L₁ ⟶ L₂) : Comma L₂ R ⥤ Comma L₁ R where
```

Comment:
I wonder if these should be redefined with the new definition? Or maybe even as abbrevs so that the new instances can apply as well?

Reply:
I saw that in the PR message just now. Oops.